lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -30,18 +30,18 @@ export default function FilePickerApp({
   const {
     authToken,
     authUrl,
-    courseId,
-    enableLmsFilePicker = false,
-    formAction,
-    formFields,
-    googleClientId,
-    googleDeveloperKey,
-    customCanvasApiDomain,
-    lmsUrl,
-    ltiLaunchUrl,
+    filePicker: {
+      formAction,
+      formFields,
+      canvas: { enabled: enableCanvasFilePicker, ltiLaunchUrl, courseId },
+      google: {
+        clientId: googleClientId,
+        developerKey: googleDeveloperKey,
+        origin: topLevelLmsUrl,
```

Comment:
Why isn't this `googleOrigin` or something directly mapped from the name?

Reply:
`topLevelLmsUrl` is just what the variable is already called in the frontend, and I'm minimising how many modifications I make to the frontend code. The frontend code is fairly incomprehensible, especially its tests that randomly fail dozens at a time when you make innocent changes that don't actually break anything. Nonetheless, it's pretty easy to rename this variable: https://github.com/hypothesis/lms/pull/1584/commits/ca0f1ee0ac27ed420df3a94da0e414b667c9eb3d